src/chunk.c

```diff
@@ -1831,12 +1840,126 @@ ts_chunk_get_by_id(int32 id, int16 num_constraints, bool fail_if_not_found)
 						   fail_if_not_found);
 }
 
+static ScanTupleResult
+chunk_form_tuple_found(TupleInfo *ti, void *data)
+{
+	FormData_chunk *form = data;
+
+	memcpy(form, GETSTRUCT(ti->tuple), sizeof(FormData_chunk));
+
+	return SCAN_DONE;
+}
+
+static bool
+chunk_get_form(int32 chunk_id, FormData_chunk *form, bool missing_ok)
+{
+	ScanKeyData scankey[1];
+	int num_found;
+	bool found;
+
+	/*
+	 * Perform an index scan on chunk id.
+	 */
+	ScanKeyInit(&scankey[0], Anum_chunk_idx_id, BTEqualStrategyNumber, F_INT4EQ, chunk_id);
+
+	num_found = chunk_scan_internal(CHUNK_ID_INDEX,
+									scankey,
+									1,
+									chunk_form_tuple_found,
+									form,
+									0,
+									ForwardScanDirection,
+									AccessShareLock,
+									CurrentMemoryContext);
+
+	found = (num_found == 1 && !form->dropped);
+
+	Assert(num_found == 0 || found);
```

Comment:
It looks like we'll always hit this assertion if there's a dropped chunk, is that the intention?

Reply:
No, I guess not. Let me take another look.